asdf/asdf.py

```diff
@@ -1344,7 +1344,57 @@ def write_to(
             ``asdf.get_config().array_inline_threshold``.
 
         """
-
+        with config_context() as config:
+            if "all_array_storage" in kwargs:
+                config.all_array_storage = kwargs["all_array_storage"]
+            if "all_array_compression" in kwargs:
+                config.all_array_compression = kwargs["all_array_compression"]
+            if "compression_kwargs" in kwargs:
+                config.all_array_compression_kwargs = kwargs["compression_kwargs"]
```

Comment:
The tests will pass without these assignments but `_find_used_blocks` (which called `reserve_blocks`) won't be aware of the `all_array_*` settings without assigning them here (this is not important for any converters except for ndarray, which does not yet exist). These assignments might also not be needed if the block manager rework results in the removal of `reserve_blocks` (which I think should be one outcome). I'm happy to remove these assignments or leave them in as they're more of an attempt to prevent a possible future bug (which we're not testing for as we don't yet have an ndarray converter).